validation/src/main/scala/hmda/validation/dsl/PredicateGeo.scala

```diff
@@ -0,0 +1,114 @@
+package hmda.validation.dsl
+
+import hmda.model.census.{ CBSAMetroMicroLookup, CBSATractLookup }
+import hmda.model.fi.lar.Geography
+
+object PredicateGeo {
+  val cbsaTracts = CBSATractLookup.values
+
+  val cbsaMetroMicro = CBSAMetroMicroLookup.values
+
+  val validMsaCombinationSet = cbsaTracts.map { cbsa =>
+    (cbsa.geoIdMsa, cbsa.state, cbsa.county, cbsa.tractDecimal)
+  }.toSet
+
+  val validMdCombinationSet = cbsaTracts.map { cbsa =>
+    (cbsa.metDivFp, cbsa.state, cbsa.county, cbsa.tractDecimal)
+  }.toSet
+
+  val validMsaCombinationSetNoTract = cbsaTracts.map { cbsa =>
+    (cbsa.geoIdMsa, cbsa.state, cbsa.county)
+  }.toSet
+
+  val validMdCombinationSetNoTract = cbsaTracts.map { cbsa =>
+    (cbsa.metDivFp, cbsa.state, cbsa.county)
+  }.toSet
+
+  val hasMsaSet = cbsaTracts.map { cbsa =>
+    (cbsa.geoIdMsa, cbsa.state, cbsa.county)
+  }.filter(x => x._1 != "").map { x =>
+    (x._2, x._3)
+  }.toSet
+
+  val hasMdSet = cbsaTracts.map { cbsa =>
+    (cbsa.metDivFp, cbsa.state, cbsa.county)
+  }.filter(x => x._1 != "").map { x =>
+    (x._2, x._3)
+  }.toSet
```

Comment:
`hasMsaSet` and `hasMdSet` seem unused. Leftovers from a previous approach, or something else?